src/Aspire.Dashboard/Model/ResourceStateViewModel.cs

```diff
@@ -70,7 +70,7 @@ private static (Icon icon, Color color) GetStateIcon(ResourceViewModel resource)
             icon = new Icons.Filled.Size16.Circle();
             color = Color.Info;
         }
-        else if (resource.HealthStatus is not HealthStatus.Healthy)
+        else if (resource.HealthStatus is not HealthStatus.Healthy && string.IsNullOrEmpty(resource.StateStyle))
```

Comment:
Instead of adding the state style check here, what about moving the state style check starting on line 78 ahead?

Reply:
Yeah that makes more sense